docs/modules/kaolin.ops.spc.rst

```diff
@@ -0,0 +1,151 @@
+structure defined by the octree. 
+The levels of the octree form a nested hierarchy 
+of cells within a :math:`2^l\times 2^l\times 2^l` partition of the unit cube.
+Every octree node corresponds to a specific cell whose smallest 
+coordinate serves as the position of the node. 
+
+Given :math:`O`, we generate the :math:`n \times 3` tensor :math:`P` that contains the 
+coordinates of octree nodes. Following the breath-first layout of the octree, the points 
+of :math:`P` are enumerated by levels in Morton order (hyperlink?). 
+Levels 0 through 8 of a structured point cloud teapot model are illustrated below:
+
+.. image:: ../img/spcTeapot.png
+
+
+A batch of octree tensors are represented by :math:`O` as a :math:`k \times 1` tensor of bytes,
+together with a :math:`b \times 1` tensor :math:`lengths`. Since :math:`O` uses packed batching,
+:math:`k` is equal to the sum the elements of :math:`lengths`.
+
+>>> import torch
+>>> import numpy as np
+>>> import kaolin.ops.spc as spc
+...
+>>> teapot = np.load("teapot.npz")
+>>> octree = torch.from_numpy(teapot["octree"], device=torch.device('cuda:0'))
+>>> lengths = torch.tensor([len(octree)], dtype=torch.int32)  
+
+The first step in generating the point hierarchy tensor is to call
+
+>>> pyramid, exsum = spc.scan_octrees(octree, lengths)
+
+The tensor :attr:`pyramid` is a tensor that contains layout 
+information for each Structured Point Cloud. 
+For the teapot example
+
+>>> pyramid[0]
+tensor([[     1,     4,    24,    108,    422,   1748,   7120,  28208, 111743,      0],
+        [     0,     1,     5,     29,    137,    559,   2307,   9427,  37635, 149378]], 
+        dtype=torch.int32)
+
+The first row of :attr:`pyramid[0]` indicates the number of points in each level of the octree,
+the second row constains the offset to each level. Note that the final dimension of
+:attr:`pyramid` is equal to two more than the octree level. In this case, the last dimension is 10,
+so the octree level must be 8. Since level 9 does not exist, :attr:`pyramid[0,0,9]` contains 0.
+However the offset to level 9 :attr:`pyramid[0,1,9]` corresponds to the number of points in
+all levels of the heierarchy. 
+
+The tensor
+:attr:`exsum` is the exclusive sum of the bit counts of each :attr:`octree` byte. This tensor
+is used by many Structured Point Cloud functions; avoiding its recomputation will improve
+performace. Next, these tensors are input to
+
+>>> points = spc.decode_points(octree, pyramid, exsum)
+
+The tensor :attr:`points` contains the batches of point hierachies.
+
+Structured point cloud data can be derived from multiple sources. Using mesh_to_spc() will
+convert a triangle mesh model into octree tensor :math:`O`. We can also sonstruct :math:`O`
+from unstructured point cloud data (coming soon?). Or from an implicit function 
+(need a cuda call back). 
+
+Feature Data
+............
+
+We associate feature data with levels in the point hierarchy. This might be a tensor
```

Comment:
That's a little bit too verbose, needs an example I think.

Reply:
Also, this part -- about what can be stored, and how it is retrieved -- belongs at the very top of documentation.